### ISSUE TITLE
npc skin <player>

### ISSUE THREAD

**Issue by AshourAziz** [CitizensDev/Citizens2#691]
/npc skin <player> -p doesn't get the latest skin of the player online.

Example, set the NPC skin of the player and use the -p flag. Player changes there skin. relogs with the new skin, try setting a NEW npc with /npc skin <player> -p it'll not reflect the new skin, rather the old skin.
Restart the server, wait awhile have the person login then do the command and it works fine.

Not the _biggest_ bug, but if you've got a running server with players and are trying to do this with well over 75 NPC's it can be a hassle to wait 75 days and 75 restarts to finally update every skin....

V2.0.17 build  1328

**Comment by fullwall** [comment]
This may be due to Mojang's caching API... will have to investigate this.

**Comment by Sarianna82** [comment]
I'm having the same problem... trying to set a bunch of skins for NPCs on a server I'm building, and it's duplicating them all to the first one I set with the -p flag, even if my own skin has cycled through several options in the meantime.  Definitely causing some major snags in my server build, unless I want a bunch of NPC clones running around.  :-(

**Comment by fullwall** [comment]
Duplicate - may be fixed in latest build anyway

**Comment by heckabecca** [comment]
The issue still persists. :( Myabe you guys could add in where the skins are saved in the plugin folder....that way we could just paste the skin.png's into the folder itself as an option...

@sarianna82 I'm in the exact same boat!

**Comment by mcmonkey4eva** [comment]
@heckabecca Unfortunately, Mojang's setup does not permit using skins that aren't associated with a player account.

**Comment by heckabecca** [comment]
@mcmonkey4eva Bummer...

I found a workaround.

I was having the issue applying skins to my NPCs. The issue was on MC Version 1.9.2, Spigot server, using Citizens 2 2.0.17. I updated my server to 1.9.4 and tried the latest plugin build, 2.0.18 (build#1369) http://ci.citizensnpcs.co/job/Citizens2/. Still didn't work as it should. HOWEVER...

....I found a workaround that works for me!

Steps To Apply NPC Skin:

Log Off of Server
Stop Server
Change player skin in Minecraft profile
Start Server
Log On to Server
Select npc and use commands:
If changing an already applied skin, first use: /npc skin reset
Apply new skin: /npc skin USERNAME -p
Goodluck!! And hopefully this gets resolved soon so its not such a hassle to apply a skin.
